Feature add typescript support

Sarsum opened the pull request.

Sarsum committed:
Add docusaurus typescript support

Sarsum committed:
Switched from jsx to tsx :)

byRoadrunner committed:
Added Typescript support for ESLint

Sarsum committed:
Fixed eslint and prettier warnings

GiantTreeLP commented on .eslintrc.json:
```diff
@@ -13,6 +13,11 @@
+    },
+    {
+      "files": [".tsx", ".ts"],
+      "parser": "eslint-mdx",
```
Is this the correct parser?

Sarsum replied:
@byRoadrunner ?

GiantTreeLP commented on package.json:
```diff
@@ -31,13 +31,22 @@
     "eslint-plugin-mdx": "^1.14.1",
     "eslint-plugin-react": "^7.24.0",
-    "prettier": "2.3.2"
+    "prettier": "2.3.2",
```
Is there a reason for a fixed version?

Sarsum replied:
@byRoadrunner ?

Sarsum committed:
Merge branch 'development' of https://github.com/CloudNetService/website into feature/typescript-support

 Conflicts:
	package.json
	yarn.lock

Sarsum committed:
Merged origin/development

byRoadrunner committed:
Rework eslint and prettier (#64)

* Reworked eslint and prettier
thanks to @Sarsum for fixing the typescript issues

* Fixed lint-action errors:
- for create event
- for runs triggered by external repositories

* Fix DeepCode warning for missing key attribute

Co-authored-by: Sarsum <sarsum@cloudnetservice.eu>

Sarsum committed:
Merge branch 'development' of https://github.com/CloudNetService/website into feature/typescript-support
